Clarify that attribute requirement levels apply to instrumentation libraries (#3289)
open-telemetry/opentelemetry-specification: CHANGELOG.md

### Semantic Conventions

> - Clarify that attribute requirement levels apply to instrumentation. And that,
>   because users can transform their telemetry in a number of ways, these requirement
>   levels cannot be relied on by telemetry consumers.

Author: ```suggestion
- Clarify that attribute requirement levels apply to the instrumentation library
```
